### ISSUE TITLE
linter: false positive on eslint-plugin-jest(expect-expect)

### ISSUE THREAD

**Issue by yunarch** [oxc-project/oxc#12869]
### What version of Oxlint are you using?

1.10.0

### What command did you run?

oxlint

### What does your `.oxlintrc.json` config file look like?

A minimum reproducible config to see the false positive:
```jsonc
// .oxlintrc.json
{
  "$schema": "https://raw.githubusercontent.com/oxc-project/oxc/main/npm/oxlint/configuration_schema.json",
  "plugins": ["vitest"],
  "overrides": [
    {
      "files": [
        "**/__tests__/**/*.{js,jsx,cjs,cjsx,mjs,mjsx,ts,tsx,cts,ctsx,mts,mtjsx}",
        "**/*.spec.{js,jsx,cjs,cjsx,mjs,mjsx,ts,tsx,cts,ctsx,mts,mtjsx}",
        "**/*.test.{js,jsx,cjs,cjsx,mjs,mjsx,ts,tsx,cts,ctsx,mts,mtjsx}",
        "**/*.bench.{js,jsx,cjs,cjsx,mjs,mjsx,ts,tsx,cts,ctsx,mts,mtjsx}",
        "**/*.benchmark.{js,jsx,cjs,cjsx,mjs,mjsx,ts,tsx,cts,ctsx,mts,mtjsx}"
      ],
      "rules": {
        "vitest/expect-expect": "error"
      }
    }
  ]
}
```

### What happened?

After updating to `1.10.0`, the following test triggers the `expect-expect` rule, reporting that the test has no assertions:
```ts
it('example test', async () => {
    await Promise.all([
      expect(
        new Promise((resolve) => {
          resolve(1);
        })
      ).resolves.toBe(1),
      expect(
        new Promise((_, reject) => {
          reject(new Error('Failed'));
        })
      ).rejects.toThrowError('Failed'),
    ]);
  });
```

However, the test clearly contains two assertions. In previous version `1.9.0` this code did not produce any errors. So it seems like this could be a regression or a change in how the rule interprets assertions.

For comparison, this rewritten version is accepted without issue:
```
  it('example test', async () => {
    await expect(
      new Promise((resolve) => {
        resolve(1);
      })
    ).resolves.toBe(1);
    await expect(
      new Promise((_, reject) => {
        reject(new Error('Failed'));
      })
    ).rejects.toThrowError('Failed');
  });
```

I think both versions are valid tests, at least they pass without problems and behave identically in terms of assertions. I understand the first version may be more difficult to analyze statically, but since it worked previously, I'm wondering if this is an unintentional regression in `1.10.0`

**Comment by camc314** [comment]
Note to self, the regression is in c0e224a19aa05c1890c76e8abcd66fd6435a9f46

**Comment by kennidenni** [comment]
@camc314 Hi! I have the same issue, still in `1.11.0`. 

`eslint-plugin-jest(expect-expect): Test has no assertions`

```
    it('test', async () => {
        const array = [1];

        for (const element of array) {
            expect(element).toBe(1);
        }
    });
```

**Comment by camc314** [comment]
thanks, see #12905
